app/code/community/Bolt/Boltpay/Model/Cron.php

```diff
@@ -44,4 +44,44 @@ public function cleanupQuotes() {
 
         $connection->query($sql);
     }
+
+    /**
+     * After a pre-auth pending order has existed for 15 minutes or more, we remove it from the system.
+     * At this point, Bolt should have called the "failed_payment" webhook to do this. We are catching
+     * the anomaly cases where this mechanism fails.
+     */
+    public function cleanupOrders() {
```

Comment:
should we have top-level try-catch just as a safe mechanism? like what happens if this some logic fails? we have no access to server-side log

Reply:
There is an individual try/catch per remove order attempt, but let's add a catch-all just in case for abnormal failures.